### PR TITLE
noticed thread safety in files

### PR DESCRIPTION
**Pull request by n-hutton**

### CODE REVIEW

File: libs/core/src/logging.cpp [hunk 1]
```diff
@@ -171,9 +171,12 @@ LogRegistry::LogRegistry() = default;
 
 void LogRegistry::Log(LogLevel level, char const *name, std::string &&message)
 {
-  if (level < global_level_)
   {
-    return;
+    FETCH_LOCK(lock_);
+    if (level < global_level_)
```
Comment (ejfitzgerald) [inline libs/core/src/logging.cpp]: I think that global log level should be set using an atomic instead

File: libs/muddle/src/router.cpp [hunk 2]
```diff
@@ -654,8 +654,8 @@ void Router::SetKademliaRouting(bool enable)
  */
 Router::Handle Router::LookupRandomHandle(Packet::RawAddress const & /*address*/) const
 {
-  static std::random_device rd;
-  static std::mt19937       rng(rd());
+  thread_local std::random_device rd;
+  thread_local std::mt19937       rng(rd());
```
Comment (ejfitzgerald) [inline libs/muddle/src/router.cpp:658]: 👍

**Review by ejfitzgerald — CHANGES_REQUESTED** [review]

File: libs/muddle/src/rbc.cpp [hunk 3]
```diff
@@ -534,6 +534,11 @@ void RBC::Deliver(SerialisedMessage const &msg, uint32_t sender_index)
   deliver_msg_callback_(miner_id, msg);
   lock_.lock();
```
Comment (bipll) [inline libs/muddle/src/rbc.cpp:535]: Here, unlocking/locking is done manually. Wouldn't it be safer to use RAII, like with `lock_guard` (but inverse)?

Comment (n-hutton) [reply libs/muddle/src/rbc.cpp:535]: yes, that would be better - I will make the change in another PR that I am working on currently

Comment (bipll) [reply libs/muddle/src/rbc.cpp:535]: May I suggest UnlockGuard for the name? :)

**Review by wkaluza — APPROVED** [review]

File: libs/muddle/src/rbc.cpp (hunk 3 above)
Comment (n-hutton) [reply libs/muddle/src/rbc.cpp:535]: sure :)

**Pull request merged**